core/server/proxy/src/main/java/alluxio/proxy/s3/CompleteMultipartUploadHandler.java

```diff
@@ -82,74 +83,103 @@ public CompleteMultipartUploadHandler(final FileSystem fs, final String baseUri)
-    final String object = s.substring(s.indexOf(AlluxioURI.SEPARATOR) + 1);
-    final String uploadId = request.getParameter("uploadId");
-    LOG.debug("(bucket: {}, object: {}, uploadId: {}) queuing task...",
-        bucket, object, uploadId);
-
-    // Set headers before getting committed when flushing whitespaces
-    httpServletResponse.setContentType(MediaType.APPLICATION_XML);
-
-    Future<CompleteMultipartUploadResult> future =
-        mExecutor.submit(new CompleteMultipartUploadTask(mFileSystem, bucket, object, uploadId,
-            IOUtils.toString(request.getReader())));
-    if (mKeepAliveEnabled) {
-      // Set status before getting committed when flushing whitespaces
-      httpServletResponse.setStatus(HttpServletResponse.SC_OK);
-      long sleepMs = 1000;
-      while (!future.isDone()) {
-        LOG.debug("(bucket: {}, object: {}, uploadId: {}) sleeping for {}ms...",
-            bucket, object, uploadId, sleepMs);
-        try {
-          Thread.sleep(sleepMs);
-        } catch (InterruptedException e) {
-          LOG.error(e.toString());
+    try {
+      if (!s.startsWith(mS3Prefix)) {
+        return;
+      }
+      // Build log message capturing the request details
+      StringBuilder sb = new StringBuilder();
+      sb.append("Alluxio S3 API received ");
+      sb.append(request.getMethod());
+      sb.append(" request: URI=");
+      sb.append(s);
+      if (request.getQueryString() != null) { sb.append("?").append(request.getQueryString()); }
+      sb.append(" User=");
+      String user = S3RestServiceHandler.getUserFromAuthorization(
+          request.getHeader("Authorization"));
+      if (user == null) { user = "N/A"; }
+      sb.append(user);
+      if (LOG.isDebugEnabled() && request.getHeaderNames() != null) {
+        // Using "DEBUG" log level to indicate verbosity of this message,
+        // but we keep it printed at the "INFO" level
+        sb.append(" Headers=");
+        Map<String, String> headerMap = new HashMap<>();
+        for (String headerName : Collections.list(request.getHeaderNames())) {
+          headerMap.put(headerName, request.getHeader(headerName));
         }
-        // TODO(czhu): figure out how to send whitespace characters while still
-        // returning a correct status code
-        // - calling getWriter().flush() commits the response (headers, status code, etc.)
-        // - https://docs.oracle.com/javaee/7/api/javax/servlet/ServletResponse.html#getWriter--
-        // periodically sends white space characters to keep the connection from timing out
-        LOG.debug("(bucket: {}, object: {}, uploadId: {}) sending whitespace...",
-            bucket, object, uploadId);
-        httpServletResponse.getWriter().print(" ");
-        httpServletResponse.getWriter().flush();
-        sleepMs = Math.min(2 * sleepMs, mKeepAliveTime);
+        sb.append(headerMap);
       }
-    } // otherwise we perform a blocking call on future.get()
+      LOG.info(sb.toString());
```

Comment:
maybe we do need the audit log of S3 client API.

Reply:
There is a PR working on that here: https://github.com/Alluxio/alluxio/pull/15811